pkg/cloud/services/s3/s3.go

```diff
@@ -173,27 +173,38 @@ func (s *Service) Delete(m *scope.MachineScope) error {
 	bucket := s.bucketName()
 	key := s.bootstrapDataKey(m)
 
-	s.scope.Info("Deleting object", "bucket_name", bucket, "key", key)
-
-	_, err := s.S3Client.DeleteObject(&s3.DeleteObjectInput{
```

Comment:
Another thought on Deletion, would be good to have the bucket overall with an object lifecycle so that every objects gets automatically deleted after 1h (or more?). This way we have some guarantees from S3 that the objects aren't going to be staying around.

Reply:
With this PR, I'm trying to eliminate the log spam around these actions. I haven't noticed any issues with objects sticking around since this code path is called every time a machine is reconciled after initial creation.

Not that I'm opposed to the addition of a lifecycle policy on the bucket, but I also don't want to expand the scope of the PR beyond a clean up to adding a new feature as well.